unliftio/src/UnliftIO/Async.hs

```diff
@@ -308,6 +327,247 @@ replicateConcurrently i m = withRunInIO $ \run -> A.replicateConcurrently i (run
 replicateConcurrently_ :: MonadUnliftIO m => Int -> m a -> m ()
 replicateConcurrently_ i m = withRunInIO $ \run -> A.replicateConcurrently_ i (run m)
 
+-- | Like 'mapConcurrently' from async, but instead of one thread per
+-- element, it does pooling from a set of threads. This is useful in
+-- scenarios where resource consumption is bounded and for use cases
+-- where too many concurrent tasks aren't allowed.
+--
+-- === __Example usage__
+--
+-- @
+-- import Say
+-- 
+-- action :: Int -> IO Int
+-- action n = do
+--   tid <- myThreadId
+--   sayString $ show tid
+--   threadDelay (2 * 10^6) -- 2 seconds
+--   return n
+-- 
+-- main :: IO ()
+-- main = do
+--   yx \<- pooledMapConcurrentlyN 5 (\\x -\> action x) [1..5]
+--   print yx
+-- @
+--
+-- On executing you can see that five threads have been spawned:
+-- 
+-- @
+-- \$ ./pool
+-- ThreadId 36
+-- ThreadId 38
+-- ThreadId 40
+-- ThreadId 42
+-- ThreadId 44
+-- [1,2,3,4,5]
+-- @
+--
+--
+-- Let's modify the above program such that the threads is less than
```

Comment:
I am not a native English speaker, but that doesn't sound totally right to me, i.e. threads is plural. Maybe better:
> Let's modify the above program such that there are less threads than...